pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -138,6 +138,37 @@ func (t ResolvedPipelineTask) IsCustomTask() bool {
 	return t.CustomTask
 }
 
+// getReason returns the latest reason if the run has completed successfully
+// If the PipelineTask has a Matrix, getReason returns the failure reason for any failure
```

Comment:
NIT: It looks like it returns the reason for the first failure met in the loop (`range`), so maybe it could say
"returns the reason for one of the failures"